### DISCUSSION TITLE
[DONE] Скачивание файлов архивом через execSHH

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#220] [Есть идея, а что если...]
С помощью linkToFileDownload скачиваю большое кол-во файлов.
Выбираю 50 файлов ,скачивает только 14.
Выбираю 20 файлов ,скачивает только 17.

Как скачать архивом используя execShh?

**Comment by ttm-alexeypolunin** [comment]
На следующей неделе мы добавим параметр `zip: "file_name"` в `linkToFileDownload` тк ограничение на количество одновременно скачиваемых файлов — это ограничение браузера. Также стало понятно, что linkToFileDownload все равно обрабатывается PHP и мы можем легко запаковать в zip.

**Reply by Makhmut2445** [reply]
Ожидаем.

**Reply by ttm-alexeypolunin** [reply]
**Готово в 3.7.47.4**

```
= : linkToFileDownload(files: $get_content; zip: "full_archive")

get_content: listReplace(list: $remove; action: "filestring" = $val; value: "value")
	
		val: fileGetContent(file: $#value[filestring])
	
	remove: rowKeysRemove(row: $replace; key: "size"; recursive: true)
		replace: rowKeysReplace(row: $train; from: json`["ext","file"]`; to: json`["type","filestring"]`; recursive: true)
			train: listTrain(list: $sel)
				sel: selectList(table: $#ntn; field: 'file_field')
```

**Reply by Makhmut2445** [reply]
Выбираю 100 строк, шестеренка крутится 1 минуту 20 секунд и выдаёт ошибку
А 80 файлом могу скачать.
![image](https://user-images.githubusercontent.com/103171127/191765096-f6d07187-9a09-43e9-b5d9-a4c6137a377a.png)
С чем это может быть связано?

**Reply by Makhmut2445** [reply]
Пожалуйста, посмотрите с чем это может быть связано? Как устранить эту проблему?

**Reply by ttm-alexeypolunin** [reply]
Неизвестно
